gammapy/datasets/map.py

```diff
@@ -407,10 +407,13 @@ def npred_signal(self, model=None):
         """
         npred_total = Map.from_geom(self._geom, dtype=float)
 
-        for evaluator in self.evaluators.values():
-            if model is evaluator.model:
-                return evaluator.compute_npred()
+        if model is not None:
+            try:
+                return self.evaluators[model].compute_npred()
+            except:
```

Comment:
It's important to always specify the kind of error to catch like `except KeyError`. Also the new error message you give is not really an improvement (it missed which key is not contained). My proposal would be to just keep the `KeyError`, that is raised if the `model_name` does  not exist.